### PR TITLE
feat: centralized bridge

### PR DESCRIPTION
**Pull request by mariocao**
This PR simplifies the current Witnet Ethereum Bridge in pursuit of a significant decrease of gas cost and developer experience while interacting with the Witnet contracts.

This PR also removes unused contracts which are still available at the `legacy` branch. :sweat_smile:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,71 +57,44 @@ The `WitnetRequestsBoard` contract provides the following methods:
...
-  - _description_: check if a request has been accepted into Witnet.
-  Contracts depending on Witnet should not start their main business logic (e.g. receiving value from third parties) before this method returns `true`.
+- **witnetCheckRequestResolved**:
+  - _description_: check if a request has been resolved by Witnet.
   - _inputs_:
-    - *_id*: the unique identifier of the data request.
+    - *_id*: the sequential identifier of a request that has been previously sent to the WitnetRequestBoard.
   - _output_:
-    - boolean telling if the request has been already accepted or not.
+    - aboolean telling if the request has been already resolved or not.
```
Comment (lrubiorod) [inline README.md:97]: typo in aboolean

File: contracts/WitnetRequestBoard.sol [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        witnet = msg.sender;
+        committee = _committee;
+        // Insert an empty request so as to initialize the requests array with length > 0
+        DataRequest memory request;
+        requests.push(request);
+    }
+
+    /// @dev Posts a data request into the WRB in expectation that it will be relayed and resolved in Witnet with a total reward that equals to msg.value.
+    /// @param _requestAddress The request contract address which includes the request bytecode.
+    /// @return The unique identifier of the data request.
```
Comment (lrubiorod) [inline contracts/WitnetRequestBoard.sol:94]: In other parts of the code you changed unique by secuential

File: contracts/WitnetRequestBoard.sol [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+            );
+            requests[_id].gasPrice = tx.gasprice;
+        }
+
+        // Update data request reward
+        requests[_id].reward = newReward;
+    }
+
+    /// @dev Reports the result of a data request in Witnet.
+    /// @param _id The unique identifier of the data request.
```
Comment (lrubiorod) [inline contracts/WitnetRequestBoard.sol:155]: the same

File: test/helpers/WitnetRequestBoardTestHelper.sol [hunk 4]
```diff
@@ -51,49 +43,44 @@ contract WitnetRequestsBoardV1 is WitnetRequestsBoardInterface {
...
     requests.push(request);
-    repFactor = _repFactor;
+    committee = _committee;
   }
 
   /// @dev Posts a data request into the WRB in expectation that it will be relayed and resolved in Witnet with a total reward that equals to msg.value.
   /// @param _requestAddress The request contract address which includes the request bytecode.
-  /// @param _inclusionReward The amount of value that will be detracted from the transaction value and reserved for rewarding the reporting of the inclusion of the data request.
-  /// @param _tallyReward The amount of value that will be detracted from the transaction value and reserved for rewarding the reporting of the final result (aka tally) of the data request.
   /// @return The unique identifier of the data request.
```
Comment (lrubiorod) [inline test/helpers/WitnetRequestBoardTestHelper.sol:58]: unique or sequential?

Comment (mariocao) [reply test/helpers/WitnetRequestBoardTestHelper.sol:58]: both! :smile: if they are sequential and they do not overflow, they are unique. I will harmonize the docs. Thx

File: contracts/WitnetRequestBoard.sol (hunk 2 above)
Comment (mariocao) [reply contracts/WitnetRequestBoard.sol:94]: I change `unique` for `sequential` as there were more occurrences

**Review by lrubiorod — APPROVED** [review]

**Pull request merged**